lightning/restore/restore.go

```diff
@@ -1742,45 +1829,54 @@ func (cr *chunkRestore) deliverLoop(
 			}
 		}
 
-		// Write KVs into the engine
-		start := time.Now()
+		// in local mode, we should write these checkpoint after engine flushed
+		// we are allowed to save checkpoint when the disk quota state moved to "importing"
+		// since all engines are flushed.
+		canSaveCheckpoint := !rc.isLocalBackend() || atomic.LoadInt32(&rc.diskQuotaState) == diskQuotaStateImporting
```

Comment:
If the current state is `diskQuotaStateImporting`, then the unsafe import and reset routine should hold the diskQuotaLock, then the RLock here should be blocked.  And after all the unsafe import tasks are finished sometime later, the disk quote state should be switched to normal. Then the write rows call will write fresh data to local storage without being flushed, thus the save chunk checkpoint is not safe.

Reply:
I think it's better just call `saveCheckpoint(rc, t, engineID, cr.chunk)` if current status is diskQuotaStateImporting before call WriteRows below